src/goto-symex/goto_symex.cpp

```diff
@@ -670,6 +678,73 @@ bool goto_symext::constant_propagate_delete_char_at(
   return true;
 }
 
+bool goto_symext::constant_propagate_case_change(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1,
+  bool to_upper)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  array_exprt string_data = s_data_opt->get();
+
+  auto &operands = string_data.operands();
+  for(auto &operand : operands)
+  {
+    auto &constant_value = to_constant_expr(operand);
+    unsigned char character = numeric_cast_v<int>(constant_value);
```

Comment:
:no_entry_sign: should use `numeric_cast_v<unsigned char>` or if unavailable `narrow`